solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java

```diff
@@ -635,12 +644,36 @@ private static SolrDocumentList getDocsFromXmlResponse(
 
   /**
    * returns a random SolrClient -- either a CloudSolrClient, or an HttpSolrClient pointed at a node
-   * in our cluster
+   * in our cluster We have different CLIENTS based on their wt setting.
+   */
+  public static SolrClient getRandomClient(Random rand) {
+    int numClients = CLIENTS.size();
+    int idx = TestUtil.nextInt(rand, 0, numClients);
+    if (idx == numClients) {
+      // Return the CloudSolrClient, it only uses javabin writerType.
+      return COLLECTION_CLIENT;
+    } else {
+      // Grabbing the first
+      int rnd = random().nextInt(CLIENTS.get(idx).values().toArray().length);
+      return (SolrClient) CLIENTS.get(idx).values().toArray()[rnd];
```

Comment:
Seems very convoluted... no need to create an array twice. Not even sure why need to create an array?